// openai/src/lib/ChatCompletionStreamingRunner.ts
import {
  type ChatCompletionChunk,
  type ChatCompletionCreateParamsStreaming,
} from 'openai/src/resources/chat/completions';
import { RunnerOptions, type AbstractChatCompletionRunnerEvents } from 'openai/src/lib/AbstractChatCompletionRunner';
import { type ReadableStream } from 'openai/src/internal/shim-types';
import { RunnableTools, type BaseFunctionsArgs } from 'openai/src/lib/RunnableFunction';
import { ChatCompletionSnapshot, ChatCompletionStream } from 'openai/src/lib/ChatCompletionStream';
import OpenAI from 'openai/src';
import { AutoParseableTool } from 'openai/src/lib/parser';

export interface ChatCompletionStreamEvents extends AbstractChatCompletionRunnerEvents {
  content: (contentDelta: string, contentSnapshot: string) => void;
  chunk: (chunk: ChatCompletionChunk, snapshot: ChatCompletionSnapshot) => void;
}

export type ChatCompletionStreamingToolRunnerParams<FunctionsArgs extends BaseFunctionsArgs> = Omit<
  ChatCompletionCreateParamsStreaming,
  'tools'
> & {
  tools: RunnableTools<FunctionsArgs> | AutoParseableTool<any, true>[];
};

export class ChatCompletionStreamingRunner<ParsedT = null>
  extends ChatCompletionStream<ParsedT>
  implements AsyncIterable<ChatCompletionChunk>
{
  static override fromReadableStream(stream: ReadableStream): ChatCompletionStreamingRunner<null> {
    const runner = new ChatCompletionStreamingRunner(null);
    runner._run(() => runner._fromReadableStream(stream));
    return runner;
  }

  static runTools<T extends (string | object)[], ParsedT = null>(
    client: OpenAI,
    params: ChatCompletionStreamingToolRunnerParams<T>,
    options?: RunnerOptions,
  ): ChatCompletionStreamingRunner<ParsedT> {
    const runner = new ChatCompletionStreamingRunner<ParsedT>(
      // @ts-expect-error TODO these types are incompatible
      params,
    );
    const opts = {
      ...options,
      headers: { ...options?.headers, 'X-Stainless-Helper-Method': 'runTools' },
    };
    runner._run(() => runner._runTools(client, params, opts));
    return runner;
  }
}
